DOCS-453 - Metrics Anomaly option (#4577)
SumoLogic/sumologic-documentation: docs/alerts/monitors/create-monitor.md

To create a monitor from the [Metrics Explorer](/docs/metrics/metrics-queries/metrics-explorer/), follow the steps below:

> 1. Open the **Metrics Explorer**:
>    * [**Classic UI**](/docs/get-started/sumo-logic-ui-classic). From Sumo Logic home, click **Metrics**.<br/><img src={useBaseUrl('img/monitors/home-metrics.png')} alt="Screenshot of the Sumo Logic home page with rectangle around the Metrics icon" style={{border: '1px solid gray'}} width="300"/>

Author: ```suggestion
1. Open the **Metrics Explorer**:
   * [**Classic UI**](/docs/get-started/sumo-logic-ui-classic). From Sumo Logic home, click **Metrics**.<br/><img src={useBaseUrl('img/alerts/monitors/home-metrics.png')} alt="Screenshot of the Sumo Logic home page with rectangle around the Metrics icon" style={{border: '1px solid gray'}} width="200"/>
```

1. The same threshold will also be applied to the histogram chart.

:::note

> The same threshold translating functionality supports to [Opening the Alerts Response Page in the Metrics Explorer](/docs/alerts/monitors/alert-response/#translating-thresholds) and [Opening Monitor in the Metrics Explorer](/docs/alerts/monitors/settings/#view-in-metrics-explorer).

Reviewer: ```suggestion
The same threshold translating functionality supports [opening the Alerts Response Page in the Metrics Explorer](/docs/alerts/monitors/alert-response/#translating-thresholds) and [opening  a monitor in the Metrics Explorer](/docs/alerts/monitors/settings/#view-in-metrics-explorer).
```

1. The same threshold will also be applied to the histogram chart.

:::note

> The same threshold translating functionality supports [opening the Alerts Response Page in the Metrics Explorer](/docs/alerts/monitors/alert-response/#translating-thresholds) and [opening a monitor in the Metrics Explorer](/docs/alerts/monitors/settings/#view-in-metrics-explorer).

Author: ```suggestion
The same threshold translating functionality supports [opening the alert response page in the Metrics Explorer](/docs/alerts/monitors/alert-response/#translating-thresholds) and [opening a monitor in the Metrics Explorer](/docs/alerts/monitors/settings/#view-in-metrics-explorer).
```

Note that the same threshold translating functionality supports to [Opening Alerts Response Page in the Metrics Explorer](/docs/alerts/monitors/alert-response/#translating-thresholds) and [Opening Monitor in the Metrics Explorer](/docs/alerts/monitors/settings/#view-in-metrics-explorer).
:::

> <!-- This will be consolidated and moved to the appropriate parts of the doc after

Author: ```suggestion
<!-- This will be consolidated and moved to the appropriate parts of the doc after https://github.com/SumoLogic/sumologic-documentation/pull/4593
```

### Anomaly

Anomaly detection leverages machine learning to identify unusual behavior and suspicious patterns by establishing baselines for normal activity. This [*AI-driven alerting*](https://www.youtube.com/watch?v=nMRoYb1YCfg) system uses historical data to minimize false positives and alerts you to deviations.

* **Model-driven detection**. Machine learning models create accurate baselines, eliminating guesswork and noise.
* **AutoML**. The system self-tunes with seasonality detection, minimizing user intervention and adjusting for recurring patterns to reduce false positives.
* **User-defined sensitivity**. Users set alert sensitivity and thresholds, providing context to filter out noise.
* **One-click playbook assignment**. Monitors automatically link to [Sumo Logic Automation Service playbooks](#automated-playbooks), expediting incident response.
* **Auto-diagnosis and recovery**. The Automation Service handles diagnosis and resolution, closing the loop from alert to recovery.
* **Customizable detection**. Advanced rules, like "Cluster anomalies," allow detection based on multiple data points exceeding thresholds within a set time frame.

> If you want to trigger alerts on outlier direction rather than anomaly detection, select **Anomaly** and enable **Use Outlier**.<br/><img src={useBaseUrl('img/monitors/monitor-detector-types-for-anomaly.png')} alt="Screenshot of the Monitor Type and Detection Method options in Sumo Logic's 'New Monitor' setup page. Logs is selected as the Monitor Type, and Anomaly is selected as the Detection Method. There is an option to use Outlier detection, which is currently toggled off." width="300"/>

Reviewer 2: I know outlier monitors are something we're moving away from in favor of anomalies, I don't know if we'd want to indicate that in documentation... I guess that would be something @banant would have input on.

Author: @banant , can you speak to Andrew's feedback?

<img src={useBaseUrl('img/monitors/metrics-query.png')} alt="metrics query.png" style={{border: '1px solid gray'}} width="600" />

#### Prerequisites

To fully leverage AI-driven alerts for metrics monitors, you'll need:

* **Automation Service**. Required for linking playbooks to metrics-based monitors.

> * **Metrics data**. Metrics data must be sent to Sumo Logic for anomaly detection.

Reviewer 2: It might be helpful to suggest the amount of data required - I think consistent 30d data would be optimal, but I'm unsure what the minimum is. I can check with the team if we think it would be good to add in.

Author: @banant , can you speak to Andrew's feedback?

> * **Metrics aggregation**. Queries that return multiple time series should be aggregated (e.g., using `sum` or `avg` operators) before applying anomaly detection.

Reviewer: ```suggestion
* **Metrics aggregation**. Queries that return multiple time series should be aggregated (for example, using `sum` or `avg` operators) before applying anomaly detection.
```

> * **Metrics aggregation**. Queries that return multiple time series should be aggregated (for example, using `sum` or `avg` operators) before applying anomaly detection.

Reviewer 2: This will be required for all queries, not just queries that return multiple time series.

Author: thanks! changed to...

* **Metrics aggregation**. Queries should be aggregated (for example, using `sum` or `avg` operators) before applying anomaly detection.

Examples:

> * **Cluster anomalies detection**. A user configures alerts for anomalies when 5 out of 10 data points in a 10-minute window exceed the baseline, allowing for precision in spiky environments.

Author: Good callout. I think I'll change it to "volatile". Does that sound ok?

> * **Cluster anomalies detection**. A user configures alerts for anomalies when 5 out of 10 data points in a 10-minute window exceed the baseline, allowing for precision in spiky environments.

Author: ```suggestion
* **Cluster anomalies detection**. A user configures alerts for anomalies when 5 out of 10 data points in a 10-minute window exceed the baseline, allowing for precision in volatile environments.
```

* **Automating resolution with playbooks**. A playbook responds to CPU usage anomalies by gathering logs, notifying teams, and restarting affected servers.

:::info Limitations

> Anomaly detection applies to one time series at a time. Multi-time series queries must be aggregated before detection.

Reviewer 2: We'll require aggregation to be applied at the end of all metric anomaly monitors trigger queries

Author: thank you! changed to "Anomaly detection applies to one time series at a time. All metrics anomaly monitor trigger queries must have aggregation applied at the end of the query before detection."
